### PR TITLE
Flow preparations for adding fund manager contract.

### PR DESCRIPTION
**Pull request by markolazic01**
Following PR contains main bridge changes: 
- Hardhat config optimization runs - 200
- Codecov.yml node version change 10.x -> 12.x
- Removal of releaseTokensTimelockPassed & approveWithdrawalAndTransferFunds functions
- ReleaseTokensByMaintainer adaptation to fundManager
- Event removal, new relevant comments
- Following test changes

### CODE REVIEW

File: contracts/ChainportMainBridge.sol [hunk 1]
```diff
@@ -169,105 +150,24 @@ contract ChainportMainBridge is Initializable, ChainportMiddleware {
...
-        require(isSignatureUsed[signature] == false, "Error: Signature already used");
-        isSignatureUsed[signature] = true;
+        require(fundManager != address(0), "Error: Fund manager not set.");
+        IERC20(token).safeTransfer(fundManager, amount);
 
-        bytes32 nonceHash = keccak256(abi.encodePacked("releaseTokens", nonce));
-        require(!isNonceUsed[nonceHash], "Error: Nonce already used.");
-        isNonceUsed[nonceHash] = true;
+        // Increase nonce to track function executions
+        functionNameToNonce["releaseTokensByMaintainer"] = functionNameToNonce["releaseTokensByMaintainer"].add(1);
```
Comment (4tal) [inline contracts/ChainportMainBridge.sol:168]: @markolazic01 change it to get it from the contract call. functionNameToNonce["releaseTokensByMaintainer"].add(1); not incrementing.

File: contracts/ChainportMainBridge.sol [hunk 2]
```diff
@@ -169,105 +150,24 @@ contract ChainportMainBridge is Initializable, ChainportMiddleware {
...
     }
 
+    // Function to transfer funds to fundManager contract
     function releaseTokensByMaintainer(
-        bytes memory signature,
         address token,
-        uint256 amount,
-        address beneficiary,
-        uint256 nonce
+        uint256 amount
```
Comment (4tal) [inline contracts/ChainportMainBridge.sol:158]: @markolazic01 We need to allow sending nonce.

File: contracts/ChainportMainBridge.sol [hunk 3]
```diff
@@ -38,39 +38,27 @@ contract ChainportMainBridge is Initializable, ChainportMiddleware {
...
     event NetworkActivated(uint256 networkId);
     event NetworkDeactivated(uint256 networkId);
-
     event TokensDeposited(address tokenAddress, address issuer, uint256 amount, uint256 networkId);
-
     event PathPauseStateChanged(address tokenAddress, string functionName, bool isPaused);
-
     event BridgeFreezed(bool isFrozen);
+    event FundManagerChanged(address newFundManager);
+    event FundsRebalanced(address target, address token, uint256 amount);
```
Comment (4tal) [inline contracts/ChainportMainBridge.sol:59]: How can I get max nonce per function?

Comment (markolazic01) [reply contracts/ChainportMainBridge.sol:59]: Mapping functionNameToNonce is public so it acts like a getter function - just call it.

File: contracts/ChainportMainBridge.sol (hunk 2 above)
Comment (markolazic01) [reply contracts/ChainportMainBridge.sol:158]: What do you mean by sending nonce?

File: contracts/ChainportMainBridge.sol (hunk 1 above)
Comment (markolazic01) [reply contracts/ChainportMainBridge.sol:168]: It is getting it from the contract call and then incrementing.

Comment (4tal) [reply contracts/ChainportMainBridge.sol:168]: Why need to increment on contract?

File: contracts/ChainportMainBridge.sol (hunk 2 above)
Comment (4tal) [reply contracts/ChainportMainBridge.sol:158]: as argument to this function

File: contracts/ChainportMainBridge.sol (hunk 3 above)
Comment (4tal) [reply contracts/ChainportMainBridge.sol:59]: Perfect.

File: test/TestMainBridge.js [hunk 4]
```diff
@@ -651,5 +477,18 @@ describe("Main Bridge Test", function () {
                 expect(await mainBridgeInstance.isPathPaused(token.address, "depositTokens")).to.be.true;
             });
         });
+
+        describe("Set cold wallet function test", () => {
```
[the remark this answers is not in the document]
Comment (markolazic01) [reply test/TestMainBridge.js]: Fixed

File: test/TestMainBridge.js [hunk 5]
```diff
@@ -248,9 +246,7 @@ describe("Main Bridge Test", function () {
             let newSafetyThreshold = 50;
 
             it("Should set safety threshold (by congress)", async function () {
-                await expect(mainBridgeInstance.connect(chainportCongress).setThreshold(newSafetyThreshold))
-                    .to.emit(mainBridgeInstance, 'SafetyThresholdChanged')
-                    .withArgs(newSafetyThreshold);
+                await mainBridgeInstance.connect(chainportCongress).setThreshold(newSafetyThreshold);
```
[the remark this answers is not in the document]
Comment (markolazic01) [reply test/TestMainBridge.js]: removed

File: test/TestMainBridge.js [hunk 6]
```diff
@@ -226,9 +226,7 @@ describe("Main Bridge Test", function () {
             let newFreezeLength = 120;
 
             it("Should set time lock (by congress)", async function () {
-                await expect(mainBridgeInstance.connect(chainportCongress).setTimeLockLength(newFreezeLength))
-                    .to.emit(mainBridgeInstance, 'TimeLockLengthChanged')
-                    .withArgs(newFreezeLength);
+                await mainBridgeInstance.connect(chainportCongress).setTimeLockLength(newFreezeLength);
```
[the remark this answers is not in the document]
Comment (markolazic01) [reply test/TestMainBridge.js]: removed

File: test/TestMainBridge.js [hunk 7]
```diff
@@ -3,17 +3,19 @@ const { signatoryAddress, signatoryPk, createHash } = require('./testHelpers')
 
 describe("Main Bridge Test", function () {
 
-    let maintainersRegistry, maintainersRegistryInstance, mainBridge, mainBridgeInstance,
+    let maintainersRegistry, maintainersRegistryInstance, mainBridge, mainBridgeInstance, fundManager,
         validator, validatorInstance, chainportCongress, maintainer, maintainers, user1, user2, token,
         tokenAmount = 50, nonceIncrease = 1, decimals = 18, freezeLength = 60, safetyThreshold = 30;
```
[the remark this answers is not in the document]
Comment (markolazic01) [reply test/TestMainBridge.js]: removed

File: contracts/ChainportMainBridge.sol (hunk 1 above)
Comment (markolazic01) [reply contracts/ChainportMainBridge.sol:168]: We need to increment the nonce to have the id of the transaction execution just to track it - you also asked for no requirements on the nonce (no nonce check)
Anyway i can implement it just explain what exactly do we need

**Pull request merged**